Add documentation and refactor check type

aidenkeating opened the pull request:
## Motivation

Currently there is no API documentation for the security service.

## Description
Add API documentation for the security service.

There is also a small refactor of the `SecurityCheckType` module.

## Progress
- [x] API docs
- [x] Refactor SecurityCheckType
- [x] Make isRooted check consistent with the Android SDK implementation.

wtrocki commented on packages/security/src/SecurityService.ts:
```diff
@@ -1,6 +1,15 @@
 import { SecurityCheck, SecurityCheckResult } from "./deviceTrust";
 
+/**
+ * Service module for handling performing and reporting checks.
```
handling**,** performing and reporting possible security issues in mobile application. 

Worth to add info that separate plugin is required.

wtrocki commented on packages/security/src/deviceTrust/SecurityCheck.ts:
```diff
@@ -1,7 +1,18 @@
 import { SecurityCheckResult } from "./SecurityCheckResult";
 
+/**
+ * Interface for a device security single check that can be executed.
```
security single check => single security check

wtrocki commented on packages/security/src/deviceTrust/SecurityCheck.ts:
```diff
@@ -1,7 +1,18 @@
+  /**
+   * Execute the check.
+   *
+   * @returns {Promise<SecurityCheckResult>}
```
TSdocs actually generates this types so this is redundant. We can just say in normal words what is returned.

wtrocki commented on packages/security/src/deviceTrust/SecurityCheckResult.ts:
```diff
@@ -1,4 +1,19 @@
   name: string;
+
+  /**
+   * Whether the check passed or failed. A successful check means that the environment this
```
```
Whether the check passed or failed. A successful check means that the environment this application is running in is more secure than otherwise, as opposed to signalling if a certain feature was enabled
```
How about:

Flag that determines if executed check finished with success or failure.

aidenkeating commented:
@wtrocki Thanks for the quick review! Really appreciated! 😄 I think I addressed everything.

Would you mind taking a look at the commit I've added (https://github.com/aerogear/aerogear-js-sdk/pull/81/commits/18c373d8a7112fcaf5d548d28414919c2b84c217). It makes the rooted check consistent with the Android SDK in name and functionality (it fails when the device is rooted).

https://github.com/aerogear/aerogear-android-sdk/blob/master/auth/src/main/java/org/aerogear/mobile/security/checks/NonRootedCheck.java

wtrocki commented:
@aidenkeating Checked. Really good job! 

![](https://photos-2.dropbox.com/t/2/AADxjDAcA1zX6DI78rK-riPPsUaThmgQ92Qia8ZKXz7DOw/12/235552519/png/32x32/3/1525874400/0/2/bitmoji-20180417103026.png/EJvNx8wBGLMFIAcoBw/wT88zOV1-xIQcuboBiHl5AJN2y9l7NTW-1iItFRoFgw?dl=0&preserve_transparency=1&size=2048x1536&size_mode=3)
